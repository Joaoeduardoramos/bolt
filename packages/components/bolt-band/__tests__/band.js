import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../band.schema.yml'));
const {
  size,
  theme,
  tag,
  full_bleed: fullBleed,
} = schema.properties;

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

const timeout = 60000;

describe('<bolt-band> Component', () => {
  afterAll(async () => {
    await stopTwigRenderer();
  }, timeout);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await renderTwig('@bolt-components-band/band.twig', {
      content: 'This is a band.',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  size.enum.forEach(async sizeChoice => {
    test(`Size variations: ${sizeChoice}`, async () => {
      const results = await renderTwig('@bolt-components-band/band.twig', {
        size: sizeChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  theme.enum.forEach(async themeChoice => {
    test(`Theme variations: ${themeChoice}`, async () => {
      const results = await renderTwig('@bolt-components-band/band.twig', {
        theme: themeChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  tag.enum.forEach(async tagChoice => {
    test(`Semantic tag usage: ${tagChoice}`, async () => {
      const results = await renderTwig('@bolt-components-band/band.twig', {
        tag: tagChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  fullBleed.enum.forEach(async fullBleedChoice => {
    test(`Full bleed: ${fullBleedChoice}`, async () => {
      const results = await renderTwig('@bolt-components-band/band.twig', {
        fullBleed: fullBleedChoice,
        content: 'This is a band.',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`Pinned content`, async () => {
    const results = await renderTwig('@bolt-components-band/band.twig', {
      size: 'large',
      content: 'This is a band.',
      pinned_content: {
        upper: [
          {
            content: 'upper pinned 1',
          },
          {
            content: 'upper pinned 2',
          },
          {
            content: 'upper pinned 3',
          },
        ],
        lower: [
          {
            content: 'lower pinned 1',
          },
          {
            content: 'lower pinned 2',
          },
          {
            content: 'lower pinned 3',
          },
        ]
      }
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
