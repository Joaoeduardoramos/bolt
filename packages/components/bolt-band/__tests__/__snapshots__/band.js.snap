// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-band> Component Basic usage 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Full bleed: false 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Full bleed: true 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Pinned content 1`] = `

<bolt-band tag="div"
           size="large"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--large t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content c-bolt-band__content--pinned c-bolt-band__content--pinned-upper">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          <div class="c-bolt-band__pinned-items">
            <div class="c-bolt-band__pinned-item">
              upper pinned 1
            </div>
            <div class="c-bolt-band__pinned-item">
              upper pinned 2
            </div>
            <div class="c-bolt-band__pinned-item">
              upper pinned 3
            </div>
          </div>
        </div>
      </bolt-wrapper>
    </div>
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
    <div class="c-bolt-band__content c-bolt-band__content--pinned c-bolt-band__content--pinned-lower">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          <div class="c-bolt-band__pinned-items">
            <div class="c-bolt-band__pinned-item c-bolt-band__pinned-item--">
              lower pinned 1
            </div>
            <div class="c-bolt-band__pinned-item c-bolt-band__pinned-item--">
              lower pinned 2
            </div>
            <div class="c-bolt-band__pinned-item c-bolt-band__pinned-item--">
              lower pinned 3
            </div>
          </div>
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: article 1`] = `

<bolt-band tag="article"
           size="medium"
           theme="dark"
           full-bleed
>
  <article class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </article>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: div 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: figure 1`] = `

<bolt-band tag="figure"
           size="medium"
           theme="dark"
           full-bleed
>
  <figure class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </figure>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: footer 1`] = `

<bolt-band tag="footer"
           size="medium"
           theme="dark"
           full-bleed
>
  <footer class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </footer>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: header 1`] = `

<bolt-band tag="header"
           size="medium"
           theme="dark"
           full-bleed
>
  <header class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </header>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: nav 1`] = `

<bolt-band tag="nav"
           size="medium"
           theme="dark"
           full-bleed
>
  <nav class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </nav>
</bolt-band>

`;

exports[`<bolt-band> Component Semantic tag usage: section 1`] = `

<bolt-band tag="section"
           size="medium"
           theme="dark"
           full-bleed
>
  <section class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </section>
</bolt-band>

`;

exports[`<bolt-band> Component Size variations: large 1`] = `

<bolt-band tag="div"
           size="large"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--large t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Size variations: medium 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Size variations: none 1`] = `

<bolt-band tag="div"
           size="none"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Size variations: small 1`] = `

<bolt-band tag="div"
           size="small"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--small t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Size variations: xsmall 1`] = `

<bolt-band tag="div"
           size="xsmall"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--xsmall t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Theme variations: dark 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="dark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-dark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Theme variations: light 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="light"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-light c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Theme variations: none 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="none"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Theme variations: xdark 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="xdark"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-xdark c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;

exports[`<bolt-band> Component Theme variations: xlight 1`] = `

<bolt-band tag="div"
           size="medium"
           theme="xlight"
           full-bleed
>
  <div class="c-bolt-band c-bolt-band--medium t-bolt-xlight c-bolt-band--full-bleed">
    <div class="c-bolt-band__content">
      <bolt-wrapper size="xxlarge"
                    full="true"
                    bolt-object
      >
        <div class="o-bolt-wrapper o-bolt-wrapper--xxlarge o-bolt-wrapper--full">
          This is a band.
        </div>
      </bolt-wrapper>
    </div>
  </div>
</bolt-band>

`;
